File: GitNexus/gitnexus/test/fixtures/lang-resolution/ruby-write-access/service.rb
require_relative 'models'

class UserService
  def update_user
    user = User.new

    # Simple write (setter method call)
    user.name = "Alice"

    # Object write
    user.address = Address.new

    # Compound assignment (operator_assignment node)
    user.score += 10
  end
end

File: GitNexus/gitnexus/test/fixtures/lang-resolution/php-write-access/service.php
<?php

require_once 'models.php';

function updateUser(User $user) {
    // Simple write
    $user->name = "Alice";

    // Object write
    $user->address = new Address();

    // Static property write
    User::$count = 42;

    // Compound assignment write
    $user->name .= " Smith";
}
